tools/ld-dropout-correct/dropoutcorrect.cpp

```diff
@@ -391,21 +472,25 @@ void DropOutCorrect::findPotentialReplacementLine(const QVector<DropOutLocation>
 // Correct a dropout by copying data from a replacement line.
 void DropOutCorrect::correctDropOut(const DropOutLocation &dropOut,
                                     const Replacement &replacement, const Replacement &chromaReplacement,
-                                    QByteArray &thisFieldData, const QByteArray &otherFieldData)
+                                    QVector<QByteArray> &thisFieldData, const QVector<QByteArray> &otherFieldData,
+                                    Statistics &statistics)
 {
     if (replacement.fieldLine == -1) {
         // No correction needed
         return;
     }
 
-    const quint16 *sourceLine = reinterpret_cast<const quint16 *>(replacement.isSameField ? thisFieldData.data()
-                                                                                          : otherFieldData.data())
+    const quint16 *sourceLine = reinterpret_cast<const quint16 *>(replacement.isSameField ? thisFieldData[replacement.sourceNumber].data()
+                                                                                          : otherFieldData[replacement.sourceNumber].data())
                                 + ((replacement.fieldLine - 1) * videoParameters[0].fieldWidth);
-    quint16 *targetLine = reinterpret_cast<quint16 *>(thisFieldData.data())
+    quint16 *targetLine = reinterpret_cast<quint16 *>(thisFieldData[0].data())
                           + ((dropOut.fieldLine - 1) * videoParameters[0].fieldWidth);
 
-    if (chromaReplacement.fieldLine == -1) {
+    // Choose whole signal or just chroma replacement
+    // Never use chroma if the source of the replacement is > 0
```

Comment:
Why? It could still be coming from a *different line* in another source, so there's still a benefit in using separate chroma/luma sources.

Maybe `if (chromaReplacement.fieldLine == -1 || (dropOut.fieldLine == replacement.fieldLine) && (dropOut.fieldLine == chromaReplacement.fieldLine))` (i.e. don't bother filtering if it's coming from another source's copy of the same line)?

Reply:
Done